what-is-chalk.mdx

```diff
@@ -18,215 +18,430 @@ import { PyDiffEditor } from '@/components/Editor'
 
-Many data sources (like vendor APIs) are too slow for online use cases
-or charge a high dollar cost-per-call. Chalk lets you optimize latency
-and cost by defining declarative caching policies that are well-integrated
-throughout the system. You no longer have to manage data sources such as
-Redis, Memcached, or DynamoDB, or spend time tuning cache-warming pipelines.
+Note: We build a DAG instead of a tree to prevent redundant feature computations.
 
-Add a caching policy with one line of code in your feature definition:
+Each node in the DAG is a Chalk expression or resolver (SQL / Python):
 
-<PyDiffEditor html={addCachingCode} />
+- lambda like Chalk expression e.g. `is_expensive_purchase: bool = _.amount > 100`
+- SQL resolver that connects to an underlying data store like Athena
+- Python resolver i.e. ordinary Python functions (Pandas, httpx, or LLM calls)
 
-Optionally warm feature caches by executing resolvers on a schedule:
-
-```sql
--- resolves: User
--- source: snowflake
--- cron: 1h
-select
-    user_id as id,
-    credit_report,
-    ts
-from credit_scores
-```
+Here's an example of a generated query plan:
+
+![Chalk query plan](/img/docs-what_is_chalk-query_plan.png)
+
+### How does Chalk resolve features? What are Chalk resolvers?
+
+Chalk connects directly to your existing infrastructure and underlying data stores (Athena, Bigquery, Postgres, Iceberg catalog, etc.) with [SQL resolvers](https://docs.chalk.ai/docs/sql-resolvers) and to APIs (microservices, 3rd part clients, LLMs) with [Python resolvers.](https://docs.chalk.ai/docs/python-resolvers)
+Resolvers make it easy to integrate a wide variety of data sources, join them together, and use them in inference, while Chalk abstracts away dependency resolution and low-latency model execution.
+
+Decoupling feature logic and model execution from underlying ETL results in
+
+- faster iteration without pipeline rebuilds
+- on-demand just-in-time inference
+- reproducible and testable features with point-in-time accuracy
+
+<IntegrationTypes />
 
-Or override staleness tolerances at query time when you need fresher
-data for your models:
+### How does Chalk orchestrate and manage features?
+
+Chalk has built-in support for feature engineering workflows -- there's no need to manage Airflow or orchestrate complicated streaming flows.
+Once we've defined our features and resolvers, Chalk orchestrates them into flexible pipelines (slicing the DAG) that make both training and model execution easy.
+
+In the `get_username` function below we explicitly define input dependencies through `User.email` and specify the output feature type as `User.username`:
 
 ```py
-from chalk.client import ChalkClient
+@online
+def get_username(
+    email: User.email, # use type annotation to tell Chalk what feature to transform
```

Comment:
let's put these comments above the annotation because we're usually trying to save horizontal space so this looks kind of reasonable on mobile

Reply:
Done!